wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -611,4 +651,29 @@ private WorkspaceImpl getByKey(String key) throws NotFoundException, ServerExcep
     }
     return workspaceDao.get(wsName, namespace);
   }
+
+  private void relativizeRecipeLinks(WorkspaceConfigImpl config) {
```

Comment:
Can we move this somewhere, say, closer to DB layer (@PrePersist or something else) to avoid duplication of the same code but with different input types (DTO vs Impl)

Reply:
As far as I understand you can not do it with preperstist since it depends on `che.api` conf property.

BTW it should not be possible to start workspaces with recipes, it should go away soon